plugins/inputs/firehose/firehose_request.go

```diff
@@ -0,0 +1,140 @@
+package firehose
+
+import (
+	"encoding/base64"
+	"encoding/json"
+	"errors"
+	"fmt"
+	"net/http"
+	"time"
+
+	"github.com/influxdata/telegraf/config"
+	"github.com/influxdata/telegraf/internal"
+)
+
+type record struct {
+	EncodedData string `json:"data"`
+}
+
+type requestBody struct {
+	RequestID string   `json:"requestId"`
+	Timestamp int64    `json:"timestamp"`
+	Records   []record `json:"records"`
+}
+
+// https://docs.aws.amazon.com/firehose/latest/dev/httpdeliveryrequestresponse.html#requestformat
+type request struct {
+	req  *http.Request
+	body requestBody
+	res  response
+}
+
+// https://docs.aws.amazon.com/firehose/latest/dev/httpdeliveryrequestresponse.html#responseformat
+type response struct {
+	body       responseBody
+	statusCode int
+}
+
+type responseBody struct {
+	RequestID    string `json:"requestId"`
+	Timestamp    int64  `json:"timestamp"`
+	ErrorMessage string `json:"errorMessage,omitempty"`
+}
+
+func (r *request) authenticate(expected config.Secret) error {
+	if expected.Empty() {
+		return nil
+	}
```

Comment:
Please put a comment here
```suggestion
	// We completely switch off authentication if not 'access_key' was provided in the config, it's intended!
	if expected.Empty() {
		return nil
	}
```